core/pom.xml

```diff
@@ -247,6 +251,18 @@
                         </configuration>
                     </execution>
                 </executions>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.codehaus.groovy</groupId>
+                        <artifactId>groovy-eclipse-compiler</artifactId>
```

Comment:
Not for this PR (as definitely good to leave versions as-is while making this big change), but it would make sense to upgrade the groovy versions at some point:
* groovy-eclipse-compiler to 2.9.2-01 (from 2.9.0-01)
* groovy-eclipse-batch to 2.4.3-01 (from 2.3.4-01)
* groovy-all to 2.3.7 (from 2.3.4)

Note that https://github.com/groovy/groovy-eclipse/wiki says eclipse-compiler 2.9.1 only supports 2.3.7 (not 2.4.x), and also note there are no release notes for 2.9.2 (even though it seems like an official release, based on the numbering scheme in maven central).